### DISCUSSION TITLE
`any` kind

### DISCUSSION THREAD

**Discussion by Arpita-Jaiswal** [fastn-stack/fastn#451] [Ideas & RFCs]
We can define any variable or argument of a kind `any`. This kind can take a variable of any kind as its value.

```ftd
-- string name: Arpita
-- any data: $name
```

Here the variable `data` is of a kind `any` which takes string kind variable `name` reference. We can use this kind in functions. 

```ftd
-- void http(url, method, ...headers: any) {
string url:
string method:

<some logic here>
}
```

In the above case, the header is the `rest operator` which is ensured using `...`. The rest operator enables to create functions that can take an indefinite number of arguments. The kind of acceptable value for argument is `any`.

```ftd
-- ftd.text: Click here
$on-click$: $http(url = "http://foo", method = "post", a = "foo", b = "get")
```

We will have a special type called `any`:

```ftd
-- any.integer foo: 10
```

Here we have created a variable `foo` or type `any`, and its initial value is `integer 10`.

To read `any` type we need `match`:

P1Script:

```ftd
-- match $foo:

-- ftd.text: integer
$case$: integer as x

-- ftd.text: percent length
$case$: length.percent as p

-- ftd.text: anything else
$case$: _

-- end: match
```

FTDScript:

```ftd
-- ftd.decimal: {
   match (foo) {
      decimal(p) {
         p
      }
      ftd.length.percent(p) {
         p.value
      }
     _ { 0.0 }
  }
}
```

**Comment by amitu** [comment]
@Arpita-Jaiswal also mention the match statement to work with values of any type.
